Version file management

phase opened the pull request:
File name conflicts can happen if they're all in the same version.

I tested with creating a project, downloading the version, renaming the
project, downloading the version again, and deleting the project.
Everything was good.

Fixes #326
Closes #327

I'm working on a script to fix the prod file paths. Don't merge until it's in.

phase committed:
Store versions in their own directories

File name conflicts can happen if they're all in the same version.

I tested with creating a project, downloading the version, renaming the
project, downloading the version again, and deleting the project.
Everything was good.

Fixes #326
Closes #327

Signed-off-by: Jadon Fowler <jadonflower@gmail.com>

phase committed:
Clean up scripts folder

I added the hash to the fix_file_paths script so you can easily tell
what commit you need to run it after.

Signed-off-by: Jadon Fowler <jadonflower@gmail.com>

kashike commented on app/db/impl/access/ProjectBase.scala:
```diff
@@ -149,11 +150,12 @@ class ProjectBase(override val service: ModelService,
 
     channel.remove()
 
+    val p = this.fileManager.getProjectDir(proj.ownerName, proj.name).resolve(channel.name)
```
nitpick: please don't use a single character for a variable name

phase replied:
that's a temp, I wanted to see if that was being called

phase committed:
Fix one letter variable name

Forgot this when testing.

Signed-off-by: Jadon Fowler <jadonflower@gmail.com>

phase committed:
Add script to fix version files and signatures

The old script was fixed too.

Fixes #328

Signed-off-by: Jadon Fowler <jadonflower@gmail.com>

phase commented:
Closes #328 too.

:tada: :tada: :tada: :tada: 

<sup><sub>Anything else you guys want to break while I'm here?</sub></sup>

phase committed:
Fix Version/Channel File Deletion

There were some weird quirks with it but it should work now.

Signed-off-by: Jadon Fowler <jadonflower@gmail.com>

phase commented:
I fixed up some issues @felixoi notified me of.

progwml6 commented on scripts/fix_file_paths_b725a7c.py3:
```diff
@@ -0,0 +1,83 @@
+      # This probably means someone uploaded a version that conflicted with another
+      # version and it got overwritten. I shouldn't be deleted because it might match
+      # another version in the DB
+      println("ERROR: " + old_file_path + " doesn't have the hash " + hash)
```
we are checking the MD5's if this doesn't match the current version it is one of the versions affected by the bug and needs to be removed from the DB

phase replied:
Done in the latest commit

progwml6 commented on scripts/fix_file_paths_b725a7c.py3:
```diff
@@ -0,0 +1,83 @@
+# version directories. This script will move the files
+# into separate directories.
+#
+# by @phase
```
felixoi and I worked on this one also

progwml6 reviewed (changes requested):
the newer file should have a different name like what the early version I sent you had was called, keep the hashes in them that is probably useful

phase committed:
Finish Script for Version Moving

It will prompt you if you want to delete project that don't have any
versions, since those are illegal in the DB.

Signed-off-by: Jadon Fowler <jadonflower@gmail.com>

phase committed:
Rename File Migration Script

Signed-off-by: Jadon Fowler <jadonflower@gmail.com>

phase commented:
This good to go?

progwml6 commented:
The readme needs to be fixed for the rename

phase commented:
Fixed
